rmc/utils/mpc.py

```diff
@@ -297,80 +294,97 @@ def prepare_pop_path_ht(
         blosum=blosum_ht[ht.ref, ht.alt].score,
         grantham=grantham_ht[ht.ref, ht.alt].score,
     )
-    ht = ht.checkpoint(
-        f"{TEMP_PATH_WITH_FAST_DEL}/joint_clinvar_gnomad_full.ht",
-        _read_if_exists=not overwrite_temp,
-        overwrite=overwrite_temp,
-    )
 
-    logger.info(
-        "Filtering to rows with defined annotations in non-outlier transcripts and writing out..."
-    )
-    transcripts = get_constraint_transcripts(outlier=False)
+    logger.info("Filtering to rows with defined annotations...")
     ht = ht.filter(
         hl.is_defined(ht.cadd.phred)
         & hl.is_defined(ht.blosum)
         & hl.is_defined(ht.grantham)
         & hl.is_defined(ht.oe)
         & hl.is_defined(ht.polyphen.score)
-        & hl.is_defined(ht.misbad)
-        & transcripts.contains(ht.transcript)
     )
-    ht.write(joint_clinvar_gnomad.versions[freeze].path, overwrite=overwrite_output)
-
+    ht = ht.checkpoint(
+        f"{TEMP_PATH_WITH_FAST_DEL}/joint_clinvar_gnomad_annot.ht",
+        _read_if_exists=not overwrite_temp,
+        overwrite=overwrite_temp,
+    )
 
-def run_regressions(
-    variables: List[str] = ["oe", "misbad", "polyphen"],
-    additional_variables: List[str] = ["blosum", "grantham"],
-    overwrite: bool = True,
-    freeze: int = CURRENT_FREEZE,
-) -> None:
-    """
-    Run single variable and joint regressions and pick best model.
+    def _add_misbad_and_filter_transcripts(
```

Comment:
I placed it here because if we do it this way, all the folds can use the same base table to filter to their training transcripts. Otherwise we would have to redo all the filtering steps for each fold